packages/core/README.md

```diff
@@ -1,18 +1,126 @@
 # Human Protocol Core Smart Contracts
 
-- Escrow
+## Contracts
 
-- EscrowFactory
+### Escrow
 
-- HMToken
+Escrow contract represents the entire life cycle of the job on Human Protocol. Escrow status is usually changed as follows.
 
-- Staking
+`Launched` -> `Pending` -> `Partial` -> `Paid` -> `Completed`
 
-- Reward Pool
+On the other hand, it can be cancelled anytime.
 
-- KVStore
+- `constructor(token, canceler, duration, trustedHandlers)`
 
-# Deploy contracts with proxy to a live network
+  Create a new escrow with ERC20 token for payments, canceler who receives the money back when the job is cancelled, job duration before expiry, and extra trusted handlers that act like super admin of the escrow. Initial escrow status is `Launched`.
+
+- `setup(reputationOracle, recordingOracle, reputationOracleFeePercentage, recordingOracleFeePercentage, url, hash)`
+
+  Assigns the reputation, and recording oracle to the job, with relevant fee percentages. Job manifest url, and hash is also configured at this point. This function should be called after the escrow is funded from the job launcher. The escrow is now in `Pending` status. Only trusted handlers can call this function.
+
+- `storeResults(url, hash)`
+
+  Stores intermediate results. Can be called when the escrow is in `Pending`, or `Partial` status. Trusted handlers, and recording oracle can call this function.
+
+- `bulkPayOut(recipients, amounts, url, hash, txId)`
+
+  Pay out the workers. Final result URL is recorded. If the escrow is fully paid out, escrow status is changed to `Paid`, otherwise it's changed to `Partial`. Trusted handlers, and reputation oracle can call this funciton.
```

Comment:
```suggestion
  Pay out the workers. Final result URL is recorded. If the escrow is fully paid out, escrow status is changed to `Paid`, otherwise it's changed to `Partial`. 
  > Trusted handlers, and reputation oracle can call this funciton.
```